### DISCUSSION TITLE
Search with API vs Search with Twitter Web

### DISCUSSION THREAD

**Discussion by hexiro** [plhery/node-twitter-api-v2#175] [Q&A] [answered]
hello! I'm trying to search, but i'm finding that the search query has no results while the web version of twitter does.
I'm defining a client with: (if that matters)
```ts
const client = new TwitterApi(TWITTER_BEARER_TOKEN).readWrite;
```

I'm searching with:
```ts
const results = await client.search(fixedEncodeURIComponent("steam://rungame/730"));
```

fixedEncodeURIComponent is defined as:
(code taken from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent)
```ts
function fixedEncodeURIComponent(str: string) {
    return encodeURIComponent(str).replace(/[!'()*]/g, function (c) {
        return "%" + c.charCodeAt(0).toString(16).toUpperCase();
    });
}
```

From my understanding, this is supposed to be a safer version of `encodeURIComponent` which will encode special characters although they shouldn't interfere with the URL.
I did this because I this matched the example twitter provided in their [search syntax docs](https://developer.twitter.com/en/docs/twitter-api/v1/rules-and-filtering/search-operators), while the original `encodeURIComponent` didn't.

If I don't include this, I get the error message, `"There were errors processing your request: no viable alternative at character '/' (at position 7), no viable alternative at character '/' (at position 8)"` and understandably so.

Upon doing this, no results are returned, whereas the web version does have results. I'm not sure if this is just caused by searching differences in the web version vs the API, or if I'm doing something wrong, But any guidance in the right direction will be very appreciated :)

**Comment by alkihis** [comment, marked as answer]
Hi :)

Some characters have a special meaning in Twitter Search API, including `/` and `:`.
You need to wrap them into quotes:
```ts
const results = await client.search('steam"://"rungame"/"730');
```

If your search is made to match exactly what you've specified, you can directly wrap everything into quotes:
```ts
const results = await client.search('"steam://rungame/730"');
``` 

A correct URL-encoding is already made by the lib, you don't need to do it by yourself :)

**Reply by hexiro** [reply]
cool! will try this out when i get home! I figured the lib would do the correct encoding, but it was erroring and i didn't understand why. Maybe I just read it wrong, but the twitter search syntax docs made it look like the quotes were optional.
